src/services/documentService.ts

```diff
@@ -0,0 +1,46 @@
+import { none, Option, some } from "fp-ts/lib/Option";
+import * as fs from "fs";
+import * as PdfDocument from "pdfkit";
+import * as shell from "shelljs";
+import { log } from "../utils/logger";
+
+export default class DocumentService {
+  public async generateDocument(
+    content: string,
+    documentPath: string
+  ): Promise<Option<Error>> {
+    return new Promise(resolve => {
+      const tempFilePath = `${process.hrtime().join("")}.pdf`;
+      const contract = new PdfDocument();
+      contract.text(content);
+      const stream = contract.pipe(fs.createWriteStream(tempFilePath));
+      contract.end();
+      stream.on("error", error => {
+        resolve(some(error));
+      });
+      stream.on("finish", async () => {
+        resolve(this.convertToPdfA(tempFilePath, documentPath));
+        fs.unlink(tempFilePath, err => {
+          if (err) {
+            log.error(
+              "Error attempting to delete temp file %s. %s",
+              tempFilePath,
+              err
+            );
+          }
+        });
+      });
+    });
+  }
+
+  private convertToPdfA(input: string, output: string): Promise<Option<Error>> {
+    const command = `gs \
```

Comment:
this string should be double checked: https://stackoverflow.com/questions/1779858/how-do-i-escape-a-string-for-a-shell-command-in-node (from the post it looks like a more secure way would be using `spawn`)